apps/docs/pages/docs/api-reference/components/auto-field.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: <AutoField>
+---
+
+import { ConfigPreview } from "@/docs/components/Preview";
+import { AutoField } from "@/core/components/AutoField";
+
+# \<AutoField\>
+
+Render a Puck field based on a [Field](/docs/api-reference/fields) object. Use this when building [custom fields](/docs/extending-puck/custom-fields) that need to use Puck-style fields internally.
+
+<ConfigPreview
+  label="Example"
+  componentConfig={{
+    fields: {
+      title: {
+        type: "custom",
+        render: ({ onChange, value }) => {
+          return (
+            <AutoField
+              field={{ type: "text" }}
+              onChange={onChange}
+              value={value}
+            />
+          );
+        },
+      },
+    },
+    defaultProps: {
+      title: "Hello, world",
+    },
+    render: ({ title }) => {
+      return <p style={{ margin: 0 }}>{title}</p>;
+    },
+  }}
+/>
+
+```tsx {1,4} copy
+import { Autofield } from "@measured/puck";
+
+const CustomField = ({ onChange, value }) => (
+  <AutoField field={{ type: "text" }} onChange={onChange} value={value} />
+);
+
+const config = {
+  components: {
+    Example: {
+      fields: {
+        title: {
+          type: "custom",
+          render: MyCustomField,
+        },
+      },
+    },
+  },
+};
+```
+
+## Props
+
+| Prop                    | Example                      | Type                                | Status   |
+| ----------------------- | ---------------------------- | ----------------------------------- | -------- |
+| [`field`](#field)       | `{ type: "text" }`           | [Field](/docs/api-reference/fields) | Required |
+| [`onChange`](#onchange) | `onChange("Goodbye, world")` | Function                            | Required |
+| [`value`](#value)       | `"Hello, world"`             | Any                                 | Required |
```

Comment:
Can `value` be `""`, `null` or `undefined`? Would assume it could need to be, but showing as `Required` here.